teps/0115-tekton-catalog-git-based-versioning.md

```diff
@@ -665,13 +676,49 @@ approach of tagging new releases in the Catalogs as is done in GitHub Actions.
 
 ## Future Work
 
-### Hub
+### Tekton Hub Catalog Registration
+
+Today, the Tekton Hub registers Catalogs through a [config.yaml][cy] file. We could provide an interface for users to share
+resources in the Tekton Hub directly, and improve the user experience in using the Tekton Hub. We can explore this in future work.
+
+The Tekton Hub already supports multiple Catalogs. As we move to providing multiple Catalogs in practice, we may need to 
+improve the scalability and performance of the Tekton Hub - most of this will be implementation details in the Tekton Hub.
+
+### Tekton Hub Git-Based Versioning
+
+To support git-based versioning in the Tekton Hub, the [config.yaml][cy] will take a `versioning` field under `catalogs`, which will default to `"directory"` and users
+can set it to `"git"`. For example, the entries in [config.yaml][cy] could be:
+
+```yaml
+catalogs:
+
+- name: tekton
+  org: tektoncd
+  type: community
+  provider: github
+  url: https://github.com/tektoncd/catalog
+
+- name: buildpacks
+  org: buildpacks
+  type: community
+  provider: github
+  url: https://github.com/buildpacks/tekton-integration
+  versioning: git
 
-Today, the Hub registers Catalogs through a [config.yaml][cy] file. We could provide an interface for users to share
-resources in the Hub directly, and improve the user experience in using the Hub. We can explore this in future work.
+- name: slack-notify
+  org: ebay
+  type: community
+  provider: github
+  url: https://github.com/eBay/tekton-slack-notify
+  versioning: git
 
-The Hub already supports multiple Catalogs. As we move to providing multiple Catalogs in practice, we may need to 
-improve the scalability and performance of the Hub - most of this will be implementation details in the Hub.
+- name: openshift
+  org: redhat
+  type: community
+  provider: github
+  url: https://github.com/openshift/tekton-catalog
+  versioning: git
+ ```
```

Comment:
How is this in "Future Work" if we deprecate the hub ? 🤔

Reply:
From the last WG, we have discussed that we will deprecate the current instance of Tekton Hub (https://hub.tekton.dev/), but still keep maintaining the [Tekton Hub Repo](https://github.com/tektoncd/hub) (and I think we discussed that the RedHat team is still interested in adding git-based versioning to Tekton Hub?), So users may still host their own instance of Tekton Hub.

But I'm happy to move it to the "alternative" section 😄

Reply:
Moved to "Alternatives" section